service/worker/workerdeployment/workflow.go

```diff
@@ -57,28 +69,29 @@ func Workflow(ctx workflow.Context, unsafeMaxVersion func() int, args *deploymen
 func (d *WorkflowRunner) listenToSignals(ctx workflow.Context) {
 	forceCANSignalChannel := workflow.GetSignalChannel(ctx, ForceCANSignalName)
 	syncVersionSummaryChannel := workflow.GetSignalChannel(ctx, SyncVersionSummarySignal)
-	forceCAN := false
 
-	selector := workflow.NewSelector(ctx)
-	selector.AddReceive(forceCANSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+	d.signalHandler.signalSelector.AddReceive(forceCANSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+		d.signalHandler.processingSignals++
 		c.Receive(ctx, nil)
-		forceCAN = true
+		d.forceCAN = true
+		d.signalHandler.processingSignals--
```

Comment:
let's put these lines in `defer` in case some one later `return`s from the middle.